src/main/java/it/antani/cc/impl/refinedstorage/ControllerImpl.java

```diff
@@ -43,4 +48,99 @@
 
         return new Object[] {all_items};
     }
+
+    /**
+     * Transfers the given amount of itemstack to the specified side and returns the number of transferred items.
+     *  Expected args passed from the Lua interpreter are:
+     *  String itemstackName : itemstack name to transfer
+     *  Double numberOfItems : number of items to transfer
+     *  String side : side where to transfer the item (north,south,up,down,est,west)
+     *  @Returns the number of transferred items.
+     */
+    @LuaMethod("extract_item")
+    public Object[] extractItem(TileController controller, IComputerAccess access, ILuaContext context, Object[] args) {
+
+        if (controller.getNetwork() == null)
+            return new Object[]{null, "Error, network not connected"};
+
+        if(args.length != 3)
+            return new Object[]{null,"Wrong number of arguments, expected arguments are : itemstackname,numberofitems,sideOfOutputInventory"};
+
+        // First argument: the itemstack name
+        String stackUnlocalizedName;
+        Optional<ItemStack> stack;
+        if (args[0] instanceof String) {
+            stackUnlocalizedName = (String) args[0];
+            stack = controller.getItemStorageCache().getList()
+                    .getStacks()
+                    .stream()
+                    .filter(itemStack -> itemStack.getUnlocalizedName().equals(stackUnlocalizedName))
+                    .findFirst();
+        }
+        else
+            return new Object[]{null,"First parameter must be the itemstack name you need to retrieve"};
```

Comment:
Please throw errors instead of successfully returning. The lua internal runtime exposes that as errors